cloud/scope/cluster.go

```diff
@@ -105,16 +107,32 @@ func (s *ClusterScope) Network() *infrav1.Network {
 }
 
 // PublicIPSpec returns the public IP specs.
-func (s *ClusterScope) PublicIPSpecs() []azure.PublicIPSpec {
-	return []azure.PublicIPSpec{
+func (s *ClusterScope) PublicIPSpecs() ([]azure.PublicIPSpec, error) {
+	spec := []azure.PublicIPSpec{
 		{
 			Name: azure.GenerateNodeOutboundIPName(s.ClusterName()),
 		},
-		{
-			Name:    s.Network().APIServerIP.Name,
-			DNSName: s.Network().APIServerIP.DNSName,
-		},
 	}
+
+	// If IP provided by name, expect to exist and don't precreate.
+	if s.AzureCluster.Spec.NetworkSpec.PublicIP != nil && s.AzureCluster.Spec.NetworkSpec.PublicIP.Name != "" {
```

Comment:
Do we allow PublicIP != nil but Name == ""?

Reply:
not any more 🙂